hbase-balancer/src/main/java/org/apache/hadoop/hbase/master/balancer/StochasticLoadBalancer.java

```diff
@@ -285,14 +286,12 @@ private boolean areSomeRegionReplicasColocated(BalancerClusterState c) {
     return false;
   }
 
-  private String getBalanceReason(double total, double sumMultiplier) {
+  private String getBalanceReason(double total) {
     if (total <= 0) {
-      return "(cost1*multiplier1)+(cost2*multiplier2)+...+(costn*multipliern) = " + total + " <= 0";
-    } else if (sumMultiplier <= 0) {
-      return "sumMultiplier = " + sumMultiplier + " <= 0";
+      return "(weighted sum of imbalance = " + total + " <= 0";
     } else if ((total / sumMultiplier) < minCostNeedBalance) {
-      return "[(cost1*multiplier1)+(cost2*multiplier2)+...+(costn*multipliern)]/sumMultiplier = " +
-        (total / sumMultiplier) + " <= minCostNeedBalance(" + minCostNeedBalance + ")";
+      return "(weighted average imbalance = " +
+        (total / sumMultiplier) + " < threshold (" + minCostNeedBalance + ")";
```

Comment:
Do we have a guard somewhere against all the multipliers being 0?

Reply:
If it is 0, the cost function is excluded at constructor.

Reply:
If they’re all 0 won’t sumMultiplier be 0 here?